src/db/client.ts

```diff
@@ -0,0 +1,59 @@
+import pg, { QueryResultRow } from 'pg';
+
+class DBClient {
+    
+    private _pool: pg.Pool;
+
+    constructor(){
+        this._pool = new pg.Pool({
+            user: process.env.DB_USER,
+            host: process.env.DB_HOST,
+            database: process.env.DB_NAME,
+            password: process.env.DB_PASS,
+            port: 5432,
+            idleTimeoutMillis: 3000
+        });
+    }
+
+    async withClient<T>(callback: (client: pg.PoolClient) => Promise<T>): Promise<T> {
+        const client = await this._pool.connect();
+        try {
+            return await callback(client);
+        } catch(e){
+            console.log(e);
+            throw e;
```

Comment:
One major issue here. In places where you are using this method `withClient`, you are sending a response first, and in this line, after sending a response, you are throwing an error, which will crash the application every time.

Reply:
The problem was in getCenterExpenses endpoint handler where I was setting response twice in case result count was 0. Corrected that now.


  ```
  async getCenterExpenses(request: Request, response: Response): Promise<void> {
        const centerId = request.params.center_id;

        try {
            const result = await this.getCenterExpenseQuery(centerId);

            if(result && result.rowCount === 0) {
                response.status(404).send("No expenses found");
            }

            response.status(200).json(result?.rows);
        } catch (error) {
            response.status(400).send(error);
        }
    }
```